.circleci/config.yml

```diff
@@ -0,0 +1,46 @@
+version: 2.1
+
+
+jobs:
+  build-dockerfile:
+    docker:
+      - image: cimg/base:2022.04
+    parameters:
+      image-name:
+        description: Image name
+        type: string
+      image-tag:
+        description: Image tag
+        type: string
+      target:
+        description: Dockerfile target
+        type: string
+        default: ""
+      dockerfile:
+        description: Dockerfile to use
+        type: string
+    steps:
+      - checkout
+      - setup_remote_docker:
+          version: 20.10.12
+      - run:
+          name: Build
+          command: |
+            echo "$DOCKER_PASS" | docker login -u "$DOCKER_USERNAME" --password-stdin
+            docker build -t "ethereumoptimism/<<parameters.image-name>>:<<parameters.image-tag>>" -f <<parameters.dockerfile>> <<#parameters.target>>--target <<parameters.target>><</parameters.target>> .
+            docker push "ethereumoptimism/<<parameters.image-name>>:<<parameters.image-tag>>"
+
+workflows:
+  main:
+    jobs:
+      - build-dockerfile:
+          context:
+            - optimism
+          filters:
+            branches:
+              only:
+                - optimism-prototype
```

Comment:
Should we build on every branch though? It's needed to run the integration tests prior to merging into main.

Reply:
Sure, I can make it do that.

Reply:
Done. We should turn this off if we're making too many changes to this repo - will fill Docker Hub up fast.